packages/flutter/lib/src/widgets/system_context_menu.dart

```diff
@@ -458,5 +458,65 @@ final class IOSSystemContextMenuItemLiveText extends IOSSystemContextMenuItem {
   }
 }
 
-// TODO(justinmc): Support the "custom" type.
-// https://github.com/flutter/flutter/issues/103163
+/// Creates an instance of [IOSSystemContextMenuItem] for custom action buttons
+/// defined by the developer.
+///
+/// Only supported on iOS 16.0 and above.
+///
+/// The [title] and [onPressed] callback must be provided.
+///
+/// {@tool dartpad}
+/// This example shows how to add custom menu items to the iOS system context menu.
+///
+/// ** See code in examples/api/lib/widgets/system_context_menu/system_context_menu.1.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [SystemContextMenu], a widget that can be used to display the system
+///    context menu.
+///  * [IOSSystemContextMenuItemDataCustom], which specifies the data to be sent
+///    to the platform for this button.
+@immutable
+class IOSSystemContextMenuItemCustom extends IOSSystemContextMenuItem with Diagnosticable {
+  /// Creates an instance of [IOSSystemContextMenuItemCustom].
+  const IOSSystemContextMenuItemCustom({
+    required this.title,
+    required this.onPressed,
+  });
+
+  @override
+  final String title;
+
+  /// The callback that is called when the button is pressed.
+  final VoidCallback onPressed;
+
+  @override
+  IOSSystemContextMenuItemData getData(WidgetsLocalizations localizations) {
+    return IOSSystemContextMenuItemDataCustom(
+      title: title,
+      callbackId: hashCode.toString(),
```

Comment:
Instead of passing the callbackId in here, could IOSSystemContextMenuItemData generate it itself?